Packet++/src/SmtpLayer.cpp

```diff
@@ -0,0 +1,248 @@
+
+		return static_cast<SmtpCommand>(val);
+	}
+
+	std::string SmtpRequestLayer::getCommandString() const { return getCommandInternal(); }
+
+	bool SmtpRequestLayer::setCommandOption(const std::string &value) { return setCommandOptionInternal(value); }
+
+	std::string SmtpRequestLayer::getCommandOption(bool removeEscapeCharacters) const
+	{
+		std::string option = getCommandOptionInternal();
+		if (!removeEscapeCharacters)
+		{
+			return option;
+		}
+
+		std::string optionWithEscapeChars;
+		for (char ch : option)
+		{
+			if (ch < 127 && ch > 31)
+			{
+				optionWithEscapeChars.push_back(ch);
+			}
+		}
+
+		return optionWithEscapeChars;
+	}
+
+	std::string SmtpRequestLayer::getCommandInfo(SmtpCommand code)
+	{
+		switch (code)
+		{
+		case SmtpCommand::DATA:
+			return "Starting mail body";
+		case SmtpCommand::EHLO:
+			return "Initiate conversation";
+		case SmtpCommand::EXPN:
+			return "Expand the mailing list";
+		case SmtpCommand::HELO:
+			return "Initiate conversation";
+		case SmtpCommand::HELP:
+			return "Ask information";
+		case SmtpCommand::MAIL:
+			return "Sender indication";
+		case SmtpCommand::NOOP:
+			return "No operation";
+		case SmtpCommand::QUIT:
+			return "Close conversation";
+		case SmtpCommand::RCPT:
+			return "Receiver indication";
+		case SmtpCommand::RSET:
+			return "Abort transaction";
+		case SmtpCommand::VRFY:
+			return "Identify user";
+		case SmtpCommand::STARTTLS:
+			return "Start TLS handshake";
+		case SmtpCommand::TURN:
+			return "Reverse the role of sender and receiver";
+		case SmtpCommand::SEND:
+			return "Send mail to terminal";
+		case SmtpCommand::SOML:
+			return "Send mail to terminal or to mailbox";
+		case SmtpCommand::SAML:
+			return "Send mail to terminal and mailbox";
+		case SmtpCommand::AUTH:
+			return "Authenticate client and server";
+		case SmtpCommand::ATRN:
+			return "Reverse the role of sender and receiver";
+		case SmtpCommand::BDAT:
+			return "Submit mail contents";
+		case SmtpCommand::ETRN:
+			return "Request to start SMTP queue processing";
+		case SmtpCommand::XADR:
+			return "Release status of the channel";
+		case SmtpCommand::XCIR:
+			return "Release status of the circuit checking facility";
+		case SmtpCommand::XSTA:
+			return "Release status of the number of messages in channel queues";
+		case SmtpCommand::XGEN:
+			return "Release status of whether a compiled configuration and character set are in use";
+		default:
+			return "Unknown command";
+		}
+	}
+
+	std::string SmtpRequestLayer::getCommandAsString(SmtpCommand code)
+	{
+		std::stringstream oss;
+		for (size_t idx = 0; idx < 8; ++idx)
+		{
+			char val = (uint64_t(code) >> (8 * idx)) & UINT8_MAX;
+			if (val) // Dont push if it is a null character
+			{
+				oss << val;
+			}
+		}
+		return oss.str();
+	}
+
+	std::string SmtpRequestLayer::toString() const
+	{
+		std::string cmd = getCommandString();
+		if (cmd.length())
+			return "SMTP Request: " + cmd;
+		return "SMTP Request";
```

Comment:
If we implement `getCommandString()` the way I suggested in the previous command we don't need this `if` statement and can always return `"SMTP Request: " + cmd;`

Also - let's be consistent with other protocols and format the string like this:
```
SMTP request layer, command: <CMD>
```